defaults/main.yml

```diff
@@ -48,3 +48,7 @@ os_projects: []
 # Ansible this can fail if there is no Cinder endpoint. Set this to true to
 # use the os_quota module rather than the openstack CLI to set quotas.
 os_projects_use_os_quota: false
+
+# Use Train upper constraints when running with Python 2, to avoid
+# incompatibility with newer OSC releases.
+os_projects_upper_constraints: "{% if ansible_python.version.major == '2' %}https://releases.openstack.org/constraints/upper/train{% endif %}"
```

Comment:
Strictly, is it actually about the version of python in os_projects_venv?

Reply:
Yes, but hopefully they're always the same? We use the same variable here:  https://github.com/stackhpc/ansible-role-os-openstacksdk/blob/master/tasks/main.yml#L54

This makes me wonder if this should be in ansible-role-os-openstackclient and ansible-role-os-openstacksdk directly?

Reply:
That's true. It probably should be in those roles directly, but also needs to be here since the default is overridden by the module dependency in meta/main.yml.

Reply:
I see, thanks.